src/org/mirah/builtins/string_extensions.mirah

```diff
@@ -39,4 +39,27 @@ class StringExtensions
       end
     end
   end
+  
+  # separate to to_i(), as to_i() (in order to maintain Ruby compatibility) could return a BigInteger,
+  # while to_int() always returns an int.
+  macro def to_int:int
+    quote { Integer.parseInt(`@call.target`) }
+  end
+  
+  macro def each_codepoint(block:Block)
+    target    = gensym
+    offset    = gensym
+    length    = gensym
+    codepoint = block.arguments.required(0).name.identifier
+    quote do
+      `target` = `@call.target` 
+      `offset` = 0
+      `length` = `target`.length
+      while `offset` < `length`
+        `codepoint` = `target`.codePointAt(`offset`)
+        `block.body`
+        `offset` = `offset` + Character.charCount(`codepoint`)
```

Comment:
I tried "+=" but got a compiler error. I guess
```mirah
`foo`+=bar
```
won't work for the same reason

```mirah
`foo`(bar)
```

currently does not work AFAIK.